instrumentation/java-http-client/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/httpclient/HttpClientInstrumentation.java

```diff
@@ -134,6 +134,7 @@ public static void methodExit(
         tracer().endExceptionally(context, null, throwable);
       } else {
         future = future.whenComplete(new ResponseConsumer(context));
+        future = CompletableFutureWrapper.wrap(future);
```

Comment:
Nit: you can pass the parent context in `@Advice.Local` to avoid calling `Context.current()` inside the wrapper